Fix build with the OS X 10.10 SDK:

error: cannot initialize a parameter of type 'id<NSFileManagerDelegate>' with an rvalue of type 'id<NSApplicationDelegate>'

https://sourceforge.net/p/irrlicht/bugs/462/
--- source/Irrlicht/MacOSX/AppDelegate.h	(revision 6519)
+++ source/Irrlicht/MacOSX/AppDelegate.h	(working copy)
@@ -10,7 +10,7 @@
 #import <Cocoa/Cocoa.h>
 #import "CIrrDeviceMacOSX.h"
 
-@interface AppDelegate : NSObject
+@interface AppDelegate : NSObject <NSApplicationDelegate, NSWindowDelegate>
 {
 	BOOL			_quit;
 	irr::CIrrDeviceMacOSX	*_device;
--- source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm.orig	2019-10-07 12:55:04.000000000 -0500
+++ source/Irrlicht/MacOSX/CIrrDeviceMacOSX.mm	2023-08-03 22:08:10.000000000 -0500
@@ -496,7 +496,7 @@
 		{
 			[[NSAutoreleasePool alloc] init];
 			[[NSApplication sharedApplication] activateIgnoringOtherApps:YES];
-			[NSApp setDelegate:(id<NSApplicationDelegate>)[[[AppDelegate alloc] initWithDevice:this] autorelease]];
+			[[NSApplication sharedApplication] setDelegate:[[[AppDelegate alloc] initWithDevice:this] autorelease]];
 			[NSBundle loadNibNamed:@"MainMenu" owner:[NSApp delegate]];
 			[NSApp finishLaunching];
 		}
@@ -718,7 +718,7 @@
 					if (!CreationParams.WindowId)
 					{
 						[Window center];
-						[Window setDelegate:(id<NSWindowDelegate>)[NSApp delegate]];
+						[Window setDelegate:(AppDelegate *)[NSApp delegate]];
 
 						if(CreationParams.DriverType == video::EDT_OPENGL)
 							[OGLContext setView:[Window contentView]];
